examples/guess-the-sketch/src/app.py

```diff
@@ -77,106 +77,135 @@ def agones_health():
+    encoded_image = base64.b64encode(model_response.content).decode('utf-8')
+    logger.info('Encoded guess image %s for player %s: %s', str(round), player_id, encoded_image)
+
+    player_guess[player_id][round]['guess_picture'] = encoded_image
+
+    # Generate and store the similarity score
+    # embedding_req = requests.post(
+    #     url = embedding_endpoint,
+    #     headers = headers,
+    #     json = json,
+    # )
+
+    similarity_prompt = f'''calculate the vector similarity number between '{player_prompt[oppontent_id][round]['prompt']}' and '{player_guess[player_id][round]['guess']}' as number A,
+                            Return number A in the format: A = '''
+    
+    similarity_payload = {
+        'prompt': f'''{similarity_prompt}''',
     }
 
-    for pending_id, pending_req in pending_requests.items():
-        if pending_id != request_id and pending_req['partner'] is None: 
-            # Match found!
-            logging.info(f'Match found')
-            pending_req['partner'] = request.sid
-            pending_req['partner_message'] = message
-            pending_requests[request_id]['partner'] = pending_id
-            pending_requests[request_id]['partner_message'] = pending_req['message']
-            payload_cur = {
+    similarity_response = requests.post(f'http://genai-api.genai.svc/genai/text', headers=headers, json=similarity_payload)
+    logger.info(f'Similarity response: {similarity_response.text}')
+
+    similarity_response_without_quota = similarity_response.text.strip('"')
+    score = float(similarity_response_without_quota.split()[-1])
+
+    logger.info(f'score: {score}')
+
+    player_prompt[player_id][round]['guess_score'] = score
+
+    # Generate and return summary once both players have guessed all 3 pictures
+
+    if round == game_round:
+        while len(player_guess) != 2:
+            time.sleep(0.1)
+        for player in player_guess:
+            while len(player_guess[player]) != game_round:
+                time.sleep(0.1)
+            for round in player_guess[player]:
+                while player_guess[player][round].get('guess_picture') is None:
+                    time.sleep(0.1)
+        for player in player_guess:
+            for round in player_guess[player]:
+                logger.debug('Sending guess picture of Player %s round %s to player %s', player, round, player_id)
+                try:
+                    emit('guess_response', {'image': player_guess[player][round]['guess_picture'], 'round': round}, room=player_id)
+                except:
+                    logger.exception('Error sending guess picture of Player %s round %s to player %s', player, round, player_id)
+
+@socketio.on('prompt')
+def handle_message(data):
+    player_id = request.sid
+    message = data['message']
+    round = int(data['round'])
+    logger.info('Received prompt %s from player %s: %s', str(round), player_id, message)
+
+    if round == 1:
+        player_prompt[player_id] = {
```

Comment:
you can use a [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict) here instead:

in globals:
player_prompt = defaultdict(dict)

then here it's just
player_prompt[player_id][round] = ...